### PR TITLE
created node script for generating test cases

### PR DESCRIPTION
**Pull request by jongund**
I have implemented a new test creation script based on Valarie's guidance in [issue 85](https://github.com/w3c/aria-at/issues/85) and it is also written in node, which actually makes the script much simpler.  The new script is in the scripts directory.

example usage to create the menubar test files:

```
node scripts\create-tests.js tests\menubar-editor
```

There is just one command to generate the scripts from the three CSV files in the `data` directory and javascript setup code in a `data/js` directory.

I still need to work on the validation components, but this script  does generate test files.

### CODE REVIEW

**Comment by jongund** [comment]
@mcking65 @spectranaut @mfairchild365 

I have made updates today to this pull request, so the test cases are now running in my local copy of test runner.
I will have Yohta start converting the other two examples to use the new CSV format.

I am still having trouble getting the startup scripts to work, so maybe there is a time that I can talk to Valarie about what I am doing wrong, or learn about the best way to debug these scripts are.

Jon

File: tests/menubar-editor/read-menuitemradio-in-a-group-in-submenu-reading.html [hunk 1]
```diff
@@ -12,23 +13,22 @@
   import { verifyATBehavior, displayTestPageAndInstructions } from "../resources/aria-at-harness.mjs";
 
   verifyATBehavior({
-    setup_script_description: "open 'Style/Color' menu item and move focue to the 'Black' menu item radio option",
-    setupTestPage: function setupTestPage(document) {
+    setup_script_description: "open 'Style/Color' menu item and move focue to the 'Black' menu item radio option",,
```
Comment (spectranaut) [inline tests/menubar-editor/read-menuitemradio-in-a-group-in-submenu-reading.html]: also to commas here

File: scripts/create-tests.js [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+  if (typeof test.setupScript === 'string') {
+    let setupScript = test.setupScript.trim();
+    if (setupScript.length) {
+      setupFileName = path.join(javascriptDirectory, test.setupScript + '.js');
+    }
+  }
+
+  validTaskReference(test.task);
+
+  let mode = getModeValue(test.mode);
```
[the remark this answers is not in the document]
Comment (jongund) [reply scripts/create-tests.js]: Ok I will add toLowerCase

File: scripts/create-tests.js [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>${test.title}</title>
+<link rel="author" title="${refs.author}" href="mailto:${refs.authorEmail}">
+${references}
+<script type="module">
+  import { verifyATBehavior, displayTestPageAndInstructions } from "../resources/aria-at-harness.mjs";
+
+  verifyATBehavior({
+    setup_script_description: "${test.setupScriptDescription}",
```
[the remark this answers is not in the document]
Comment (jongund) [reply scripts/create-tests.js]: I just fixed this in my most recent push.

File: tests/menubar-editor/read-menuitemradio-in-a-group-in-submenu-reading.html (hunk 1 above)
Comment (jongund) [reply tests/menubar-editor/read-menuitemradio-in-a-group-in-submenu-reading.html]: Fixed the double commas

File: scripts/create-tests.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jongund) [reply scripts/create-tests.js]: Fixed the capitalization issues

**Comment by jongund** [comment]
@spectranaut 
Can you look in the test in `tests\checkbox-gen`.   It is not running in test runner for some reason, where all the menubar-editor tests seem to work.   It has me stumped.

File: scripts/create-tests.js [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+  process.exit();
+}
+
+if (args._.length !== 1) {
+  console.log("Command expects a directory name, please supply.");
+  process.exit();
+}
+
+const validModes = ['reading', 'interaction'];
+const validAppliesTo = ['JAWS', 'NVDA', 'VoiceOver', 'Orca'];
```
[the remark this answers is not in the document]
Comment (jongund) [reply scripts/create-tests.js]: added "Screen Readers" to the list

File: scripts/create-tests.js [hunk 5]
```diff
@@ -0,0 +1,408 @@
...
+// Create AT commands file
+
+function createATCommandFile(cmds) {
+
+  const fname = path.join(testDirectory, 'commands.json');
+  let data = {};
+
+  function addCommand(task, mode, at, key) {
+
+    if (typeof key !== 'string' || key.length === 0) {
```
[the remark this answers is not in the document]
Comment (jongund) [reply scripts/create-tests.js]: I can look into this validation feature

**Comment by jongund** [comment]
@spectranaut 

Did something change with permissions for the repository, I can't push changes to this branch.  I get the following error:

```
jongunderson aria-at (update-create-test-scripts) $ git push
ERROR: Permission to w3c/aria-at.git denied to deploy key
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
jongunderson aria-at (update-create-test-scripts) $ 
```

**Comment by mcking65** [comment]
> @spectranaut
> 
> Did something change with permissions for the repository, I can't push changes to this branch. I get the following error:
> 
> ```
> jongunderson aria-at (update-create-test-scripts) $ git push
> ERROR: Permission to w3c/aria-at.git denied to deploy key
> fatal: Could not read from remote repository.
> 
> Please make sure you have the correct access rights
> and the repository exists.
> jongunderson aria-at (update-create-test-scripts) $ 
> ```

@jongund That can happen also if the network connection for your git terminal loses connectivity.

**Comment by jongund** [comment]
@spectranaut 

Seems to be a problem with this machine and that repository, I can push on other repositories on this machine and I can push to the aria-at repository on another machine.   A head scratcher, I was pushing to aria-at from this machine on Thursday.

File: scripts/create-tests.js [hunk 6]
```diff
@@ -0,0 +1,483 @@
...
+
+    if (typeof data[task] !== 'object') {
+      data[task] = {};
+    }
+
+    if (typeof data[task][mode] !== 'object') {
+      data[task][mode] = {};
+    }
+
+    if (typeof data[task][mode][at] !== 'object' ) {
```
Comment (spectranaut) [inline scripts/create-tests.js]: All of these keys should also be changed to lower case. the commands.json file should have lower case keys!

**Comment by spectranaut** [comment]
> @spectranaut
> Can you look in the test in `tests\checkbox-gen`. It is not running in test runner for some reason, where all the menubar-editor tests seem to work. It has me stumped.

I looked into this, @jongund! The problem is with the commands.json file -- no commands were found. In the commands.json creation, all the keys should be lowercased, I left a comment above.

**Comment by jongund** [comment]
@spectranaut 
Thanks for finding the problem!
I have updated the script to lowercase all the keys for the `commands.json` file.

**Pull request merged**